### ISSUE TITLE
Review the CSP before launch.

### ISSUE THREAD

**Issue by muffinresearch** [mozilla/addons-blog#126]
It would be good to look at the CSP as a whole before launch and review if there are opportunities to make it stronger.

Specific areas for consideration would be to review the use of `'self``. 

For example:

https://github.com/mozilla/addons-blog/blob/84d5b5a162bed8e626469f90ca584445ec078bcc/bin/deploy.sh#L48 uses `'self'` for script (which was copied across from the extension workshop CSP. This means the entire origin addons.mozilla.org can be used for script. It would be good to limit this further by at least using a path if we aren't going to put statics on a separate CDN.

**Comment by willdurand** [comment]
>  uses `'self'` for script (which was copied across from the extension workshop CSP. This means the entire origin addons.mozilla.org can be used for script. It would be good to limit this further by at least using a path if we aren't going to put statics on a separate CDN.

ha right, we can probably replace `'self'` with `/blog/assets/` everywhere I suppose?

**Comment by willdurand** [comment]
> It would be good to limit this further by at least using a path if we aren't going to put statics on a separate CDN.

We didn't really talk about using a CDN for statics. I guess we could but it would make the deployment process more complicated and the build step would need some more work.

**Comment by willdurand** [comment]
@muffinresearch I just checked again and that looks good to me. I've noticed the xss blog post produces a few CSP errors (which is what we want).

**Comment by willdurand** [comment]
@g-k

**Comment by willdurand** [comment]
@g-k hello, as per the RRA recommendations for this project, we would like to request an audit of the CSP and potentially the sanitization mechanism implemented. Would you be able to help? Thank you!

**Comment by g-k** [comment]
> @g-k hello, as per the RRA recommendations for this project, we would like to request an audit of the CSP and potentially the sanitization mechanism implemented. Would you be able to help? Thank you!

Happy to help! Do you have a code, diff, or a staging instance you want me to check out?

**Comment by g-k** [comment]
Thanks @willdurand!

That CSP policy looks good to me. The `unsafe-inline` `style-src` isn't ideal, but it's hard to avoid in practice and not particularly risky. Nice job using distinct paths for different asset types and DOMPurify to sanitize the content.

Interesting that the atom feed permits sameorigin framing, I wonder if RSS readers load the feed in an iframe.

As an aside, we should look into adding COOP and COEP headers to mitigate spectre-type attacks sometime.

https://web.dev/coop-coep/
https://w3c.github.io/webappsec-post-spectre-webdev/#documents-isolated

**Comment by willdurand** [comment]
Thank you.

> The `unsafe-inline` `style-src` isn't ideal, but it's hard to avoid in practice and not particularly risky

Indeed, I filed an issue some time ago for that because, as you say, it isn't ideal: https://github.com/mozilla/addons-blog/issues/79. Good to hear that it isn't super risky, though.

> Interesting that the atom feed permits sameorigin framing, I wonder if RSS readers load the feed in an iframe.

Is there a risk here? Can we mitigate this problem?

> As an aside, we should look into adding COOP and COEP headers to mitigate spectre-type attacks sometime.

TIL! Thanks, I'll read more about these headers, they could probably be useful on all AMO apps.

**Comment by g-k** [comment]
> > Interesting that the atom feed permits sameorigin framing, I wonder if RSS readers load the feed in an iframe.

> Is there a risk here? Can we mitigate this problem?

I don't think so, same origin is fine and it looks like embedding Atom feeds isn't uncommon. XFO deny or `frame-ancestors: none` CSP would be more restrictive, but if it hurts discovery then that would undermine the point of the blog.
